src/ImageSharp/Memory/Allocators/ArrayPoolMemoryAllocator.cs

```diff
@@ -12,140 +13,131 @@ namespace SixLabors.ImageSharp.Memory
     /// </summary>
     public sealed partial class ArrayPoolMemoryAllocator : MemoryAllocator
     {
-        private readonly int maxArraysPerBucketNormalPool;
-
-        private readonly int maxArraysPerBucketLargePool;
+        /// <summary>
+        /// The upper threshold to pool arrays the shared buffer. 1MB
+        /// This matches the upper pooling length of <see cref="ArrayPool{T}.Shared"/>.
+        /// </summary>
+        private const int SharedPoolThresholdInBytes = 1024 * 1024;
 
         /// <summary>
-        /// The <see cref="ArrayPool{T}"/> for small-to-medium buffers which is not kept clean.
+        /// The default value for the maximum size of pooled arrays in bytes. 2MB.
         /// </summary>
-        private ArrayPool<byte> normalArrayPool;
+        internal const int DefaultMaxArrayLengthInBytes = 2 * SharedPoolThresholdInBytes;
 
         /// <summary>
-        /// The <see cref="ArrayPool{T}"/> for huge buffers, which is not kept clean.
+        /// The default bucket count for <see cref="largeArrayPool"/>.
         /// </summary>
-        private ArrayPool<byte> largeArrayPool;
+        private const int DefaultMaxArraysPerBucket = 16;
```

Comment:
Since we utilize only one bucket now, an intensive parallel load will defer to the unmanaged stuff almost all the time. I would try benchmarking with different bucket counts.

Reply:
I'm gonna need help writing those tests and improving others to ensure they work in CI.